data_warehouse/views.py

```diff
@@ -28,3 +30,20 @@ def get_canvas_options(self, request):
         '''getting canvas-options'''
         data = CanvasOptionsService.get_canvas_options()
         return Response(data)
+
+    @action(detail=False, methods=['GET'], url_path='table-export',
+            url_name='table-export')
+    def export(self, request):
+        '''Return a xls file stream.'''
+        if 'table_type' not in request.GET:
+            raise BadRequest('请求的参数不正确。')
+        table_type = request.GET['table_type']
+        table_type = int(table_type)
+        context = {key: val for (key, val) in request.GET.items()}
+        context['request'] = request
+        ret_file_path = AggregateDataService.dispatch('table_export', context)
+        file_name = AggregateDataService.TABLE_NAME_CHOICES[
```

Comment:
I think you can provide a method to get `file_name` (or return file name together with your `ret_file_path`), then you can expose information about how it works as little as possible.